tests/network-tests/src/Api.ts

```diff
@@ -161,8 +218,18 @@ export class Api {
     )
   }
 
-  public getMemberIds(address: string): Promise<MemberId[]> {
-    return this.api.query.members.memberIdsByControllerAccountId<Vec<MemberId>>(address)
+  // Many calls in the testing framework take an account id instead of a member id when an action
+  // is intended to be in the context of the member. This function is used to do a reverse lookup.
+  // There is an underlying assumption that each member has a unique controller account even
+  // though the runtime does not place that constraint. But for the purpose of the tests we throw
+  // if that condition is found to be false to esnure the tests do not fail. As long as all memberships
+  // are created through the membership fixture this should not happen.
+  public async getMemberId(address: string): Promise<MemberId> {
+    const ids = await this.api.query.members.memberIdsByControllerAccountId<Vec<MemberId>>(address)
+    if (ids.length > 1) {
```

Comment:
Indeed we should, I chose to assert that ids.length == 1. It most closely communicates our intention.
This is a really ugly hack to enforce unique member controller. It would be much better to do a full refactoring of the fixtures to always require a memberId as a parameter rather than do a reverse lookup by account id.

Done in 7d63b04fac263910ba8f45c55730788fb37b6373